packages/autotask-client/src/bin.ts

```diff
@@ -1,41 +1,159 @@
   }
 
   try {
-    await updateCode();
-  } catch (err) {
-    console.error(`Error updating Autotask code: ${err.message}`);
+    validateId(autotaskId);
+    const client = initClient();
+    output(`\nPolling latest runs of autotask '${autotaskId}'...\n`, stream);
+    // Poll autotask runs every 2 seconds and if there are new runs, get run details and print them out.
+    let lastRun: AutotaskRunListItemResponse | undefined;
+    while (true) {
+      const newRuns = await client.listAutotaskRuns(autotaskId);
+      // If cached last run id has changed
+      if (newRuns.items[0]?.autotaskRunId !== lastRun?.autotaskRunId) {
+        lastRun = newRuns.items[0]; // cache new last run to avoid duplicates.
+
+        if (lastRun.status !== 'success') {
+          output(`\nLatest run '${lastRun.autotaskRunId}' ${lastRun.status}...`, stream);
+          lastRun = undefined; // clean up so we can check it again next time.
+        } else {
+          output(`\nLatest run '${lastRun.autotaskRunId}' ${lastRun.status}...`, stream);
+          const runDetails = await client.getAutotaskRun(lastRun.autotaskRunId);
+          // Have to make this check to satisfy Typescript.
+          if (runDetails.status === 'success') {
+            output(`\n${runDetails.decodedLogs}`, stream);
+            output(`\n------------------------------------------------------------`, stream);
+            output(`\nPolling latest runs of autotask '${autotaskId}'...\n`, stream);
+          }
+        }
+      }
+
+      await new Promise((resolve) => setTimeout(resolve, 2000));
+    }
+  } catch (error) {
+    const err = error as Error | AxiosError;
+    output(`Error on listening to Autotask runs: ${err.message}`, stream);
     process.exit(1);
   }
 }
 
+/**
+ * Utilizes the Autotask API to trigger autotask run manually.
+ */
+async function executeRun() {
+  const autotaskId = argv[3];
+  try {
+    validateId(autotaskId);
+    const client = initClient();
+    console.log(`Executing autotask run for autotask '${autotaskId}'...`);
+    // TODO: pass in params maybe? Do we have any case for that?
+    const resp = await client.runAutotask(autotaskId, {});
+    console.log(`Successfully executed autotask run for autotask '${autotaskId}'`);
+    console.log(`Run ID: ${resp.autotaskRunId}, \nStatus: ${resp.status}`);
+    console.info(`Tip: Call 'defender-autotask tail-runs ${autotaskId}' to follow the runs.`);
+  } catch (error) {
+    const err = error as Error | AxiosError;
+    console.error(`Error executing autotask run: ${err.message}`);
+    process.exit(1);
+  }
+}
+
+async function main() {
+  mandatoryParamGuard(argv[2] as Command);
+
+  switch (argv[2]) {
+    case 'update-code':
+      await updateCode();
+      break;
+    case 'tail-runs':
+      await tailRuns();
+      break;
+    case 'execute-run':
+      await executeRun();
+      break;
+    default:
+      throw new Error(`unhandled command '${argv[2]}'. Fix your paramGuard.`);
```

Comment:
Would the dev understand what "Fix your paramGuard" means? It seems like the guidance is to "Use one of the following commands..." or something like this

Reply:
This message is dedicated only to the developer working on the monorepo.
Before this error can be thrown, we have `mandatoryParamGuard(command)` called, which prints out the usage examples.

In practice, the developer who is working with the library will never see that error. Only the dev who is changing the library might encounter it and take note.

I will update the message though, as it mentions the old function name instead of `mandatoryParamGuard` :)

Reply:
I Hope [this](https://github.com/OpenZeppelin/defender-client/pull/99/commits/2de4069abbae710fc6979b207ff9ba9e6c04a989) makes more sense now.